devito/function.py

```diff
@@ -868,71 +868,45 @@ def coefficients(self):
 
```

Comment:
I can't claim I understand what's going on in this new implementation, but it looks cool